New Translation: docs/working-with-images-in-markdown.md (#282)
gatsbyjs/gatsby-pt-BR: docs/docs/working-with-images-in-markdown.md

---
title: Trabalhando com imagens dentro de Postagens e Páginas Markdown
---

> Quando estiver construindo sites compostos primeiramente de Páginas ou Posts com Markdown utilizando o Gatsby, a inserção de imagens reforça o conteúdo. Você pode adicionar imagens de diversas formas. 

Reviewer: ```suggestion
Quando estiver construindo sites compostos primeiramente de páginas ou posts com Markdown utilizando o Gatsby, a inserção de imagens reforça o conteúdo. Você pode adicionar imagens de diversas formas. 
```

## Imagens em destaque como metadados Frontmatter

Em sites como blogs, você pode querer inserir uma imagem em destaque, para que ela apareça no topo da página. Uma forma de fazer isso é utilizar o nome do arquivo da imagem em um campo frontmatter e então transformá-lo por meio do `gatsby-plugin-sharp` em uma query GraphQL.

> Essa solução presume que você já possua páginas de Markdown sendo geradas automáticamente por meio de renderizadores como `gatsby-transformer-remark` ou `gatsby-plugin-mdx`. Senão, leia [Parte 7 do Tutorial de Gatsby](/tutorial/part-seven/). Isso será desenvolvido durante o tutorial, nesse exemplo será utilizado o `gatsby-transformer-remark`.

Reviewer: ```suggestion
Essa solução presume que você já possua páginas de Markdown sendo geradas automaticamente por meio de renderizadores como `gatsby-transformer-remark` ou `gatsby-plugin-mdx`. Senão, leia a [parte 7 do tutorial de Gatsby](/tutorial/part-seven/). Isso será desenvolvido durante o tutorial, nesse exemplo será utilizado o `gatsby-transformer-remark`.
```

> Nota: Isso pode ser feito de forma similar utilizando [MDX](/docs/mdx/). Ao invés de usar nodes `markdownRemark` no GraphQL, O `Mdx` pode ser trocado e deverá funcionar também.

> Para iniciar, faça o download dos plugins do Gatsby-image, como mencionado em [Usando gatsby-image](/docs/using-gatsby-image/).

Reviewer: ```suggestion
Para iniciar, baixe os plugins do `Gatsby-image`, como mencionado em [Usando gatsby-image](/docs/using-gatsby-image/).
```

```shell
npm install --save gatsby-image gatsby-transformer-sharp gatsby-plugin-sharp
```

> Você também irá querer ter `gatsby-source-filesystem` instalado. Só então, configure os vários plugins no arquivo `gatsby-config`.

Reviewer 2: ```suggestion
Talvez você também queira ter instalado o  `gatsby-source-filesystem`. Então, faça a configuração dos vários plugins no arquivo `gatsby-config`.
```

### Configurando para imagens e posts no mesmo diretório

> Se as imagens e arquivos Markdown estiverem no mesmo diretório, a obtenção e solução das imagens poderá ser feita com o uso de apenas uma configuração. Por exemplo, se seus arquivos Markdown e imagens estão localizados em um diretório `/pages`, ambos os conteúdos serão automáticamente selecionados pelo GraphQL como parte da camada de dados do Gatsby.

Reviewer: ```suggestion
Se as imagens e arquivos Markdown estiverem no mesmo diretório, a obtenção e solução das imagens poderá ser feita com o uso de apenas uma configuração. Por exemplo, se seus arquivos Markdown e imagens estão localizados em um diretório `/pages`, ambos os conteúdos serão automaticamente selecionados pelo GraphQL como parte da camada de dados do Gatsby.
```

> Se as imagens e arquivos Markdown estiverem no mesmo diretório, a obtenção e solução das imagens poderá ser feita com o uso de apenas uma configuração. Por exemplo, se seus arquivos Markdown e imagens estão localizados em um diretório `/pages`, ambos os conteúdos serão automáticamente selecionados pelo GraphQL como parte da camada de dados do Gatsby.

Reviewer 2: ```suggestion
Se as imagens e arquivos Markdown estiverem no mesmo diretório, a utilização das imagens pode ser feita com apenas uma configuração. Por exemplo, se seus arquivos Markdown e imagens estão em um mesmo diretório `/pages`, ambos os conteúdos serão automaticamente selecionados pelo GraphQL como parte da camada de dados do Gatsby.
```

> Nota: Isso pode ser feito de forma similar utilizando [MDX](/docs/mdx/). Ao invés de usar nodes `markdownRemark` no GraphQL, O `Mdx` pode ser trocado e deverá funcionar também.

<!-- !> Note: This can be done similarly using [MDX](/docs/mdx/) as well. Instead of the `markdownRemark` nodes in GraphQL, `Mdx` can be swapped in and should work. -->

Para iniciar, faça o download dos plugins do Gatsby-image, como mencionado em [Usando gatsby-image](/docs/using-gatsby-image/).

<!-- ! To start out, download the plugins for Gatsby-image as mentioned in [Using gatsby-image](/docs/using-gatsby-image/). -->

```shell
npm install --save gatsby-image gatsby-transformer-sharp gatsby-plugin-sharp
```

Você também irá querer ter `gatsby-source-filesystem` instalado. Só então, configure os vários plugins no arquivo `gatsby-config`.
<!-- ! You will also want to have `gatsby-source-filesystem` installed. Then, configure the various plugins in the `gatsby-config` file. -->

### Configurando para imagens e posts no mesmo diretório

Se as imagens estiverem no mesmo diretório

> If your images are in the same directory as the Markdown files, sourcing and resolving the images can be done in one configuration. For example, if your Markdown pages and images are located together in a `/pages` directory, both content types will be automatically picked up by GraphQL as part of Gatsby's data layer.

Reviewer 2: Acho que tu deixou passar batido esse trecho XD

Author: NOSSA! Vou dar uma conferida aqui, peço mil perdões por isso 😪

Author: Realmente, faltou esse trecho, irei corrigir isso o mais rápido possível !

Novamente, peço desculpas por tal erro e irei evitar que isso ocorra novamente 🤗

Author: Pelo que vi, me parece que esse trecho está traduzido na versão atual do Doc.

Posso estar enganado, por favor me corrija nesse caso. 😀

    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/src/pages`, // Linha mais importante
      },
    },
  ],
}
```

Então, em um exemplo de um arquivo Markdown, adicione um campo chamado `featuredImage`:

```markdown:title=src/pages/Meus-Dogs-Favoritos.md
---
title: Meus Dogs Favoritos
featuredImage: cachorro-fofinho.png
---

> O conteudo será inserindo aqui!

Reviewer: ```suggestion
O conteúdo será inserido aqui!
```

```

> O próximo passo é incorporar os dados em um modelo usando uma query GraphQL, que será vista futuramente nesse guia.

Reviewer: ```suggestion
O próximo passo é incorporar os dados em um modelo usando uma query GraphQL, que será vista futuramente neste guia.
```

### Configurando para imagens e posts em diretórios diferentes

> Existem também ocasiões nas quais você possa querer que suas imagens estejam em um diretório diferente de seus posts ou páginas em Markdown, tal como em um diretório externo `/images`. Você pode definir isso ao especificar duas orignens distintas, uma para as páginas e outra para as imagens:

Reviewer: ```suggestion
Existem também ocasiões nas quais você possa querer que suas imagens estejam em um diretório diferente de seus posts ou páginas em Markdown, tal como em um diretório externo `/images`. Você pode definir isso ao especificar duas origens distintas, uma para as páginas e outra para as imagens:
```

> Existem também ocasiões nas quais você possa querer que suas imagens estejam em um diretório diferente de seus posts ou páginas em Markdown, tal como em um diretório externo `/images`. Você pode definir isso ao especificar duas orignens distintas, uma para as páginas e outra para as imagens:

Reviewer 2: ```suggestion
Em algumas ocasiões você as imagens e seus posts ou páginas Markdown vão estar em diretórios diferentes, como em um diretório externo `/images`. Você pode definir isso ao especificar duas origens distintas, uma para as páginas e outra para as imagens:
```

    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/src/pages`, // Linha mais importante
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/src/images`, // Linha mais importante
      },
    },
  ],
}
```

Então, no arquivo Markdown, o caminho para o campo `featuredImage` será relativo ao arquivo da página (nesse caso, na pasta `/images` que está um nível acima).

```markdown:title=src/pages/about.md
---
title: About
featuredImage: ../images/equipe-gatos.png
---

> O conteudo será inserindo aqui!

Reviewer: ```suggestion
O conteúdo será inserido aqui!
```

```

### Consultando imagens para Frontmatter

Agora que você já definiu a origem do Markdown e os dados da imagem, você poderá consultar por imagens em destaque no GraphQL. Se o caminho apontado for uma imagem, ele será transformado em um node `File` no GraphQL e então você poderá obter os dados da imagem utilizando o campo `childImageSharp`.

> Isso pode ser adicionado a uma query GraphQL em um arquivo modelo de Markdown. Neste exemplo, a [Query flúida](/docs/gatsby-image#images-that-stretch-across-a-fluid-container) é usada para deixar a imagem responsiva.

Reviewer: ```suggestion
Isso pode ser adicionado a uma query GraphQL em um arquivo modelo de Markdown. Neste exemplo, a [Query fluida](/docs/gatsby-image#images-that-stretch-across-a-fluid-container) é usada para deixar a imagem responsiva.
```

`
```

> Sua imagem em destaque deverá aparecer diretamente no topo da postagen gerada. Tada!

Reviewer: ```suggestion
Sua imagem em destaque deverá aparecer diretamente no topo da postagem gerada. Tada!
```

## Imagens Inline com `gatsby-remark-images`

> Você também pode incluir imagens diretamente no corpo do Markdown. O plugin [gatsby-remark-images](/packages/gatsby-remark-images) será util nesse caso.

Reviewer: ```suggestion
Você também pode incluir imagens diretamente no corpo do Markdown. O plugin [gatsby-remark-images](/packages/gatsby-remark-images) será útil nesse caso.
```

Inicie instalando `gatsby-remark-images` e `gatsby-plugin-sharp`.

```shell
npm install --save gatsby-remark-images gatsby-plugin-sharp
```

> Você também irá querer ter `gatsby-source-filesystem` instalado e que ele esteja apontando para o diretório onde suas imagens estarão localizadas.

Reviewer 2: ```suggestion
Tenha certeza que o `gatsby-source-filesystem` está instalado e que ele esteja apontando para o diretório onde suas imagens estarão localizadas.
```

}
```

> Com as configurações acima, você poderá utlizar a sintaxe padrão para imagens no Markdown. Elas serão processadas pelo Sharp e irão aparecem como se você tivesse as inserido em um componente `gatsby-image`.

Reviewer: ```suggestion
Com as configurações acima, você poderá utilizar a sintaxe padrão para imagens no Markdown. Elas serão processadas pelo `Sharp` e irão aparecem como se você tivesse às inserido em um componente `gatsby-image`.
```
